File: node/lambda-manual/handler.js
var coldStart = true;

exports.handler = async function(event, context) {
  try {
    return { cold_start: coldStart }
  } finally {
    coldStart = false;
  }
}

if (process.env._HANDLER) {
  const datadog = require('datadog-lambda-js');
  exports.handler = datadog.datadog(exports.handler);
}

File: python/awslambdaric/handler.py
import json
import os.path
import urllib.request

class cold: start = True

def is_extension_exists():
    return os.path.exists('/opt/extensions/datadog-agent')

def is_extension_running():
    try:
        urllib.request.urlopen('http://127.0.0.1:8124/lambda/hello')
        return True
    except Exception as e:
        print(f'error pinging extzension: [{e.__class__.__name__}] {e}')
        return False

def handler(event, context):
    try:
        return {
                'statusCode': 200,
                'body': json.dumps({
                    'cold_start': cold.start,
                    'extension_exists': is_extension_exists(),
                    'extension_running': is_extension_running(),
                }),
        }
    finally:
        cold.start = False

if __name__ == '__main__':
    print(handler({},{}))
